### PR TITLE
Merge pull request #1 from TestUpCommunity/integration_1

### PR DESCRIPTION
**Pull request by namelaowang**
Merge pull request #750 from namelaowang/feature_namelaowang

### CODE REVIEW

**Comment by namelaowang** [comment]
gists - check-if-a-gist-is-starred #626
gists - unstar-a-gist #625
gists - star-a-gist #624
gists - delete-a-gist #629
gists/comments - create-a-comment #633
gists/comments - get-a-single-comment #632
gists/comments - list-comments-on-a-gist #631
gists - list-gist-forks #628
gists - fork-a-gist #627
gists - list-gist-commits #623
gists - create-a-gist #621
gists - get-a-specific-revision-of-a-gist #620
gists - list-all-public-gists #617
gists - list-a-users-gists #616
gists - get-a-single-gist #619
gists - list-starred-gists #618

完成igists下gists.py和comment.py

**Comment by namelaowang** [comment]
#434
#436
#444
#437
#438
#439
#440
#442
#443
#464
#465
#460
#463
#462
#461
#458
#445
#455
#457
#456
#453
#454
#450
#452
#451
#448
#449
#447
#446

api/orgs/ 下新增 outside_collaborators.py，hooks.py，blocking.py，orgs.py，members.py

**Comment by namelaowang** [comment]
#431  完成需求#431

File: api/interactions/interactions.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Interactions(RestClient):
     https://developer.github.com/v3/interactions/#interactions
     """
 
-    def __init__(self, api_root_url, **kwargs):
-        super(Interactions, self).__init__(api_root_url, **kwargs)
-        self.orgs = Orgs(self.api_root_url, **kwargs)
-        self.repos = Repos(self.api_root_url, **kwargs)
+    # def __init__(self, api_root_url, **kwargs):
```
Comment (zhangting85) [inline api/interactions/interactions.py:11]: 把这个注释掉干嘛？

File: github.py [hunk 2]
```diff
@@ -12,4 +15,13 @@ def __init__(self, **kwargs):
         self.checks = Checks(self.api_root_url, **kwargs)
         self.interactions = Interactions(self.api_root_url, **kwargs)
         self.apps = Apps(self.api_root_url, **kwargs)
+        self.orgs = Orgs(self.api_root_url, **kwargs)
+
+
+if __name__ == '__main__':
+    g = Github(username="namelaowang", password ="ghl6032069")
```
Comment (zhangting85) [inline github.py:22]: 多余的调试语句

File: operations/repo.py [hunk 3]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+
+def delete_repo_from_user_by_name(github, owner, repo):
+    """
+
+    :param github: Github创建出来github对象
+    :param owner: 当前登陆的用户
+    :param repo: 要删除的repo关键字
+
+    """
+    import re
```
Comment (zhangting85) [inline operations/repo.py:52]: import语句放在了错误的位置。请放在文件最上面

File: operations/repo.py [hunk 4]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+    delete_list = []
+    result = CommonItem()
+    result.success = False
+
+    payload = {'try': 'all'}
+    # 列出当前用户的所有repos
+    response = github.repos.list_user_repos(username=owner, json=payload)
+    if response.status_code == 200:
```
Comment (zhangting85) [inline operations/repo.py:60]: 如果不等于200，那么result.success应该为false

File: operations/repo.py [hunk 5]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+
+        r = response.json()
```
Comment (zhangting85) [inline operations/repo.py:62]: 如果无法做json解析呢？你这样写就会报语法错误了。

File: operations/repo.py [hunk 6]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+
+        for i in r:
+            # 从该账户的所有repos中，匹配出所有带有"repo"关键字的repos
+            restring = re.match('.*{}.*'.format(repo), i['name'])
```
Comment (zhangting85) [inline operations/repo.py:66]: 不需要用正则表达式。

File: operations/repo.py [hunk 7]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+def delete_repo_from_user_by_name(github, owner, repo):
+    """
+
+    :param github: Github创建出来github对象
+    :param owner: 当前登陆的用户
+    :param repo: 要删除的repo关键字
+
+    """
+    import re
+    delete_list = []
```
Comment (zhangting85) [inline operations/repo.py:53]: 不要自己增加需求，并没有列表，就是删指定名字的repo

File: operations/repo.py [hunk 8]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+
+        r = response.json()
+
+        for i in r:
+            # 从该账户的所有repos中，匹配出所有带有"repo"关键字的repos
+            restring = re.match('.*{}.*'.format(repo), i['name'])
+
+            if restring == None:
+                continue
+            delete_list.append(restring.group())
```
Comment (zhangting85) [inline operations/repo.py:70]: 没有列表，不要自己加需求。

File: operations/repo.py [hunk 9]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+            response = github.repos.delete_a_repo(owner=owner, repo=delete_repo)
+
+            if response.status_code == 204:
+                result.success = True
+            else:
+                result.error = "delete repo got {},should be 204".format(str(response.status_code))
+
+        return result
+
+    return ['list the repos fail',  'https://developer.github.com/v3/repos/#list-user-repositories']
```
Comment (zhangting85) [inline operations/repo.py:82]: 错误的返回值。

File: operations/repo.py [hunk 10]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+
+
+def create_repo_and_branch_with_helloword(github, owner, name, branch,  description=None, homepage=None, private=False,
+                                          has_issues=True, has_projects=True, has_wiki=True):
+
```
Comment (zhangting85) [inline operations/repo.py:87]: 文档呢?

File: operations/repo.py [hunk 11]
```diff
@@ -37,3 +39,74 @@ def create_repo(github, name, org=None, description=None, homepage=None, private
...
+    response_repos = github.repos.create_user_repo(json =payload)
+    if response_repos.status_code == 201:
+        # 创建分支
+        response_branch = github.repos.branches.add_required_status_checks_contexts_of_protected_branch(
+            owner=owner, repo=name, branch=branch)
+        if response_branch == 200:
+            # 上传文件
+            pass
+
+        return "创建分支失败"
```
Comment (zhangting85) [inline operations/repo.py:107]: 错误的返回值。请返回commonItem

**Comment by zhangting85** [comment]
错误太多，打回修改。

**Pull request closed without merging**